common/lib/dependabot/shared_helpers.rb

```diff
@@ -258,18 +258,19 @@ def self.with_git_configured(credentials:, &_block)
       FileUtils.mkdir_p(Utils::BUMP_TMP_DIR_PATH)
 
       previous_config = ENV.fetch("GIT_CONFIG_GLOBAL", nil)
+      git_config_global_path = File.expand_path("#{SecureRandom.hex(16)}.gitconfig", Utils::BUMP_TMP_DIR_PATH)
```

Comment:
Can you add an inline comment explaining why this is needed?

Reply:
done